### PR TITLE
v2: integration tests covering routes + v2 migration

### PR DESCRIPTION
**Pull request by stegaBOB**
Stacked on top of #127. Adds an integration test suite split per-concern.

```
tests/
├── common/{mod.rs, rpc.rs}  — per-test postgres + axum harness + wiremock JSON-RPC mock
├── routing.rs       — 22 tests: empty-DB + seeded-data per endpoint; 2 trust-filter regressions (ignored, pending fix)
├── sweep.rs         — 5 tests driving sweep::run_once against mocked getMultipleAccounts
├── webhooks.rs      — 12 tests for /pda + /unverify (auth-shape + behaviour)
├── migration.rs     — 7 tests: v1 -> v2 + edge cases (idempotent, empty, NULL authority, un-used status, etc)
└── verify_smoke.rs  — 1 end-to-end /verify_sync test on mainnet, ignored, weekly cron
```

`tests/common/rpc.rs` includes byte builders for BPF Loader v3 Program / ProgramData accounts so wiremock responses round-trip through `parse_bpf_upgradeable_loader` without extra Solana deps.

`DbClient::pool()` and `sweep::run_once()` are now pub for test access; also reasonable escape hatches for future admin tooling.

`TEST_DATABASE_URL` skips testcontainers and uses a local postgres (fresh `CREATE DATABASE` per test). Otherwise testcontainers spins one up via Docker.

`src/main.rs` split into `src/lib.rs` + a thin `src/main.rs` so tests can use the crate as a library.

The verify smoke test runs via `.github/workflows/verify-smoke.yaml` on a Monday 06:00 UTC cron + `workflow_dispatch`. Slow (~10-20 min), depends on Docker Hub + GitHub + mainnet RPC.

### CODE REVIEW

**Review by De-ASI-INTERFACE — APPROVED** [review]

**Comment by stegaBOB** [comment]
Here's the stacked diff (instead of from here to main): https://github.com/solana-mobile/solana-verified-programs-api/pull/13

**Review by Rchie121 — APPROVED** [review]

**Review by Rchie121 — APPROVED** [review]

**Review by Rchie121 — CHANGES_REQUESTED** [review]
> ``Fixes #128

**Review by De-ASI-INTERFACE — APPROVED** [review]

**Review by Rchie121 — APPROVED** [review]

**Review by De-ASI-INTERFACE — APPROVED** [review]

**Review by meumar-osec — APPROVED** [review]

File: .github/workflows/verify-smoke.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-verify-smoke-${{ hashFiles('Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-verify-smoke-
+
+      - name: Install solana-verify CLI
+        run: cargo install solana-verify --locked
```
Comment (swaroop-osec) [inline .github/workflows/verify-smoke.yaml]: My only concern is that this installs `solana-verify` from crates.io without specifying a version, while production pins the version in `Dockerfile:16`.

This may not necessarily match the Docker version. It would be better to pin the workflow install to the same source/tag as production.

**Review by swaroop-osec — APPROVED** [review]
Just a minor nit, non-blocking

**Review by swaroop-osec — APPROVED** [review]
Lgtm

**Review by De-ASI-INTERFACE — APPROVED** [review]

**Review by De-ASI-INTERFACE — APPROVED** [review]

**Review by De-ASI-INTERFACE — APPROVED** [review]

**Review by De-ASI-INTERFACE — APPROVED** [review]

**Review by De-ASI-INTERFACE — APPROVED** [review]

**Review by De-ASI-INTERFACE — APPROVED** [review]

**Review by De-ASI-INTERFACE — APPROVED** [review]
